Freshworks-Samples/App-Development-Features/Advanced-Features/data_storage/custom_entities/server/server.js

```diff
@@ -0,0 +1,40 @@
+exports = {
+  events: [{
+    event: 'onTicketUpdate',
+    callback: 'onTicketUpdateHandler'
+  }, ],
+  onTicketUpdateHandler: function (args) {
+    console.log("OnTicketUpdate handler fired.");
+    let ticket = args.data.ticket;
+    console.log("Check if ticket is closed. Ticket status : ", ticket.status);
+    if (ticket.status == "5") {
+      console.log("Ticket has been closed. Updating the respective restaurant record based on the ticket ID filter");
+      let ticket_id = ticket.id;
+      let entity = $db.entity({
+        version: "v1"
+      });
+      let appointments = entity.get("appointments");
+      let restaurants = entity.get("restaurants");
+      let appointmentRecords = appointments.getAll({
+        query: {
+          ticket_id
+        }
+      });
+      appointmentRecords.then((i) => {
+        restaurants.get(i.records[0].data.restaurant_id).then((d) => {
+          console.log("Restaurant with matching ticket id ", d.record.data);
+          let uObj = d.record.data;
+          uObj.status = "3";
```

Comment:
Status with `5` or `3` is hard to read. Shall we have an ENUM for better readability.

```js
const TICKET_STATUS = {
  pending: 3,
  closed: 5
};
```

> Applicable at other such instances as well.

```suggestion
          uObj.status = TICKET_STATUS.pending;
```

Reply:
added